hyperjumptech/monika-config-generator
# feat: Add google analytics with consent

Implementing issue #63 

- by default, analytics is disallowed
- after user agree to use cookie, then the analytics is running on every route change

### Screenshot

<img width="1149" alt="Screen Shot 2021-04-27 at 12 52 15" src="https://user-images.githubusercontent.com/5974862/116192209-b7a57780-a757-11eb-9f0d-8e9b836e28ec.png">

### How to test

1. open http://localhost:3000
2. don't click on cookies consent (or click decline. It's the same)
3. see developer console on tab network > Other
4. go to other page and see there is `no` call on url `collect?v=2...` (https://www.google-analytics.com/g/collect?v=2...)
5. clear cookies (or you can open http://localhost:3000 on new incognito)
6. accept cookie consent
7. see developer console on tab network > Other
8. go to other page and see there is `a` call on url `collect?v=2...` (https://www.google-analytics.com/g/collect?v=2...)
